src/main/java/net/imglib2/labkit/models/CachedImageFactory.java

```diff
@@ -0,0 +1,22 @@
+
+package net.imglib2.labkit.models;
+
+import net.imglib2.RandomAccessibleInterval;
+import net.imglib2.cache.img.CellLoader;
+import net.imglib2.img.Img;
+import net.imglib2.img.cell.CellGrid;
+import net.imglib2.labkit.segmentation.Segmenter;
+import net.imglib2.type.NativeType;
+
+import java.util.function.Consumer;
+
+/**
+ * Factory for creating a cached image that stores segmentation/prediction
+ * results
+ */
+public interface CachedImageFactory {
+
+	<T extends NativeType<T>> Img<T> setupCachedImage(Segmenter segmenter,
```

Comment:
@tpietzsch What is the $segmenter$ parameter intended to be used for. It is never used, can it be removed?

Reply:
I use it to get
```
List< String > labels = segmenter.classNames();
```
to determine channel names and colors for Imaris (every class becomes one channel).